src/engine/sidechain/engineshoutcast.cpp

```diff
@@ -92,14 +92,33 @@ EngineShoutcast::EngineShoutcast(ConfigObject<ConfigValue>* _config)
         errorDialog(tr("Error setting non-blocking mode:"), shout_get_error(m_pShout));
         return;
     }
+
+    m_bThreadQuit = false;
+    this->moveToThread(&m_SThread);
+    connect(&m_SThread, SIGNAL(started()), this, SLOT(shoutcastThread()));
+    connect(this, SIGNAL(finished()), &m_SThread, SLOT(deleteLater()));
```

Comment:
I know there are blogs that you should not subclass QThread for various reasons. 
But they do not apply in this case. We just do not need a Qt message queue here. 

I would prefer here to do it similar to EngineSideChain and treat shoutcast as sibling to EngineSideChain. 
Instantiate it in EgineMaster and so on. 
No moveToThread and re-implement run() ...